datafusion/core/src/datasource/listing/table.rs

```diff
@@ -433,7 +433,7 @@ impl ListingTableConfig {
     /// `SchemaAdapterFactory` is set, in which case only the `SchemaAdapterFactory` will be used.
     ///
     /// See <https://github.com/apache/datafusion/issues/16800> for details on this transition.
-    pub fn with_physical_expr_adapter_factory(
+    pub fn with_expr_adapter_factory(
```

Comment:
In writing the example I noticed that this name is very long and probably too verbose.
This tones it down and brings it in line with the field name.

Reply:
I love this explanation -- I have found trying to write out examples of using APIs almost always leads to discoveries of ways to improve them for the better. Thank you!